crates/types/src/lib.rs

```diff
@@ -1527,34 +1527,26 @@ pub struct Table {
     /// The table elements' Wasm type.
     pub wasm_ty: WasmRefType,
     /// The minimum number of elements in the table.
-    pub minimum: u32,
+    pub minimum: u64,
     /// The maximum number of elements in the table.
-    pub maximum: Option<u32>,
+    pub maximum: Option<u64>,
+    /// Whether or not this is a 64-bit table.
+    pub table64: bool,
 }
 
 impl TypeTrace for Table {
     fn trace<F, E>(&self, func: &mut F) -> Result<(), E>
     where
         F: FnMut(EngineOrModuleTypeIndex) -> Result<(), E>,
     {
-        let Table {
-            wasm_ty,
-            minimum: _,
-            maximum: _,
-        } = self;
-        wasm_ty.trace(func)
+        self.wasm_ty.trace(func)
```

Comment:
Mind leaving this as it was prior? You'll be adding `table64: _` to the pattern but it helps us remember to consider updating these impls when fields are added or removed from `Table`